src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -416,65 +486,224 @@ class AresDNSResolver : public DNSResolver {
       return {reinterpret_cast<intptr_t>(this), aba_token_};
     }
 
+   protected:
+    AresRequest(grpc_pollset_set* interested_parties, AresDNSResolver* resolver,
+                intptr_t aba_token)
+        : pollset_set_(grpc_pollset_set_create()),
+          interested_parties_(interested_parties),
+          completed_(false),
+          resolver_(resolver),
+          aba_token_(aba_token) {
+      GRPC_CLOSURE_INIT(&on_dns_lookup_done_, OnDnsLookupDone, this,
+                        grpc_schedule_on_exec_ctx);
+      grpc_pollset_set_add_pollset_set(pollset_set_, interested_parties_);
+    }
+
+    // closure to call when the ares resolution request completes. Subclasses
+    // should use this as the ares callback in MakeRequestLocked()
+    grpc_closure on_dns_lookup_done_ ABSL_GUARDED_BY(mu_);
```

Comment:
Data members must be private, not protected, as per the style guide.

https://google.github.io/styleguide/cppguide.html#Access_Control

However, you can add protected accessor methods for them.